### DISCUSSION TITLE
Exception when running on an ESP8266EX

### DISCUSSION THREAD

**Discussion by catalinsanda** [arendst/Tasmota#21767] [Q&A] [answered]
Hey folks,

I'm trying to install Tasmota on an ESP8266EX with 4Mb of flash. Tried the Lite & Minimal var, both install, but crash when trying to run. 
Here are the logs, any idea what could cause this?

Thanks!

```
load 0x4010f000, len 3584, room 16
tail 0
chksum 0xb0
csum 0xb0
v3969889e
~ld

00:00:00.001 HDW: ESP8266EX
00:00:00.010 CFG: Use defaults
00:00:00.033 Project tasmota - Tasmota Version 14.1.0(release-minimal)-2_7_6(2024-06-03T11:53:06)
00:00:00.034 WARNING This version does not support persistent settings
00:00:00.289 WIF: WifiManager active for 3 minutes
00:00:00.797 HTP: Web server active on x5864 with IP address 192.168.4.1

--------------- CUT HERE FOR EXCEPTION DECODER ---------------

Exception (0):
epc1=0x4000dce5 epc2=0x00000000 epc3=0x00000000 excvaddr=0x00000000 depc=0x00000000

>>>stack>>>

ctx: cont
sp: 3ffffd50 end: 3fffffc0 offset: 0190
3ffffee0:  4021c5d8 00000000 00001388 40213529
3ffffef0:  00000000 00000000 00000000 40213db4
3fffff00:  00000000 4bc6a7f0 00000c10 3fff0380
3fffff10:  00000000 00000000 4bc6a7f0 00000000
3fffff20:  00000000 00000000 40100302 c76c8b43
3fffff30:  00000000 4bc6a7f0 40100302 c3d70a3d
3fffff40:  00000032 3ffee830 00000000 3fff0380
3fffff50:  3fffdad0 00000000 3ffee830 4021034a
3fffff60:  3fffdad0 00000000 00000c10 40210cca
3fffff70:  3fffdad0 00000000 00000c10 40210d7c
3fffff80:  00000000 00000000 00000001 40100214
3fffff90:  3fffdad0 00000000 3fff0340 3fff0380
3fffffa0:  3fffdad0 00000000 3fff0340 402198cc
3fffffb0:  feefeffe feefeffe 3ffe8538 40100de1
<<<stack<<<

--------------- CUT HERE FOR EXCEPTION DECODER ---------------

 ets Jan  8 2013,rst cause:2, boot mode:(3,2)

```

**Comment by Jason2866** [comment]
Bad power supply. If it is a dev board the onboard 3.3V LDO is to weak. Replace with a type which can deliver at least 500mA.

**Comment by catalinsanda** [comment]
While not impossible I would say it's unlikely. This is the device: https://www.aliexpress.com/p/order/index.html and I've managed to run ESP Home fine on it. Here are the logs:

```
[I][logger:034]: Log initialized
[C][safe_mode:079]: There have been 0 suspected unsuccessful boot attempts
[I][app:029]: Running through setup()...
[C][wifi:047]: Setting up WiFi...
[C][wifi:060]: Starting WiFi...
[C][wifi:061]:   Local MAC: 08:F9:E0:4F:16:E8
[D][wifi:073]: Loaded saved wifi settings: --redacted--
[D][wifi:481]: Starting scan...
[W][component:157]: Component wifi set Warning flag: scanning for networks
[D][wifi:496]: Found networks:
--- REDACTED ---
[I][wifi:312]: WiFi Connecting to '--redacted--'...
[W][wifi_esp8266:504]: Event: Disconnected ssid='--redacted--' bssid=--redacted-- reason='Handshake Failed'
[W][wifi:652]: Error while connecting to network.
[D][wifi:696]: Retrying with hidden networks...
[I][wifi:312]: WiFi Connecting to '--redacted--'...
[I][wifi:616]: WiFi Connected!
[C][wifi:427]:   Local MAC: 08:F9:E0:4F:16:E8
[C][wifi:432]:   SSID: '--redacted--'
[C][wifi:435]:   IP Address: --redacted
[C][wifi:438]:   BSSID: --redacted--
[C][wifi:440]:   Hostname: 'esphome-web-4f16e8'
[C][wifi:442]:   Signal strength: -63 dB ▂▄▆█
[C][wifi:446]:   Channel: 1
[C][wifi:447]:   Subnet: 255.255.255.0
[C][wifi:448]:   Gateway: --redacted--
[C][wifi:449]:   DNS1: 8.8.8.8
[C][wifi:450]:   DNS2: 8.8.4.4
[D][wifi:625]: Disabling AP...
[C][web_server:128]: Setting up web server...
[C][api:025]: Setting up Home Assistant API server...
[I][app:062]: setup() finished successfully!
[W][component:170]: Component wifi cleared Warning flag
[W][component:157]: Component api set Warning flag: unspecified
[I][app:100]: ESPHome version 2024.6.6 compiled on Jul  5 2024, 20:21:06
[I][app:102]: Project esphome.web version 1.0
[C][wifi:599]: WiFi:
[C][wifi:427]:   Local MAC: 08:F9:E0:4F:16:E8
[C][wifi:432]:   SSID: '--redacted--'
[C][wifi:435]:   IP Address: --redacted--
[C][wifi:438]:   BSSID: --redacted--
[C][wifi:440]:   Hostname: 'esphome-web-4f16e8'
[C][wifi:442]:   Signal strength: -63 dB ▂▄▆█
[C][wifi:446]:   Channel: 1
[C][wifi:447]:   Subnet: 255.255.255.0
[C][wifi:448]:   Gateway: --redacted--
[C][wifi:449]:   DNS1: 8.8.8.8
[C][wifi:450]:   DNS2: 8.8.4.4
[C][logger:185]: Logger:
[C][logger:186]:   Level: DEBUG
[C][logger:188]:   Log Baud Rate: 115200
[C][logger:189]:   Hardware UART: UART0
[C][captive_portal:088]: Captive Portal:
[C][web_server:173]: Web Server:
[C][web_server:174]:   Address: esphome-web.local:80
[C][mdns:115]: mDNS:
[C][mdns:116]:   Hostname: esphome-web-4f16e8
[C][esphome.ota:073]: Over-The-Air updates:
[C][esphome.ota:074]:   Address: esphome-web.local:8266
[C][esphome.ota:075]:   Version: 2
[C][safe_mode:018]: Safe Mode:
[C][safe_mode:019]:   Boot considered successful after 60 seconds
[C][safe_mode:021]:   Invoke after 10 boot attempts
[C][safe_mode:022]:   Remain in safe mode for 300 seconds
[C][api:139]: API Server:
[C][api:140]:   Address: esphome-web.local:6053
[C][api:144]:   Using noise encryption: NO
[C][improv_serial:032]: Improv Serial:
[I][safe_mode:041]: Boot seems successful; resetting boot loop counter
```

**Reply by gg23303013** [reply]
It works fine with the old tasmota12.5.bin which was built last year. After recompiling Tasmota 12.5, it doesn't work anymore. I am new. Do you know how to recompile tasmota 14.1 with last year's core?

**Comment by Jason2866** [comment]
Highly unlikely it is not the device ;-) Tasmota does run without issues on many thousands of esp8266 devices.
A small esphome firmware does not stress the power supply as the feature rich Tasmota firmware does.
So this does tell nothing about possible bad hardware. You asked for reason what causing the problem.
You should check what the reason for 99% of issues with Tasmota on esp8266 devices is in our experience.

**Reply by catalinsanda** [reply]
Thanks!, I'll power the chip directly from a 3.3v power supply and report back.

**Reply by catalinsanda** [reply]
Powered the chip directly from a RIGOL DP832 bench power supply, bypassing the LDO, and same thing happened. 
Might be an issue with the chip's flash and by sheer luck ESPHome isn't using that part so I'll stick with it for now.

**Reply by sfromis** [reply]
You might also have a device issue with a if a decoupling capacitor is bad or "missing". But yes, bad flash is also possible, including if you had a dubious power supply when doing the flashing. In the latter case, flashing with better power available *might* help.

**Comment by gg23303013** [comment]
I just bought an ESP8266EX too. It seems to have a problem with Tasmota 14.1. I tested it, and it works fine with the old Tasmota 12.5

**Reply by sfromis** [reply]
I've got quite a few Tasmota devices with the ESP8266EX chip, not seeing any trouble, with current Tasmota builds.

**Reply by gg23303013** [reply]
It is a new, cheapest module on AliExpress. I use Tasmota 12.5 temporarily because it seems stable. Hope this helps someone who bought it

**Reply by sfromis** [reply]
No, ESP8266EX is the small 5x5 mm chip, nothing more, and generally works fine with Tasmota, as long as the circuitry around it is ok, very much including how power is delivered. Issues with power is not at all uncommon when having trouble. And it is certainly possible that whatever module the chip is on is not great.

**Reply by gg23303013** [reply]
I tested again. After recompiling tasmota 12.5 it doesn't work anymore. It only works fine with the old tasmota12.5.bin which was build last year. The newly purchased nodemcu module contains the esp8266ex chip and antenna on board, not the esp12 extension, so it looks neater.

**Reply by sfromis** [reply]
That's no principal hardware difference. The ESP-12 module is merely a well-designed module with ESP8266EX under the metal shield. Hopefully your board is as well-designed, but that's a risk factor. To Tasmota, both are **exactly** the same.

**Reply by gg23303013** [reply]
I'm new. I plan to test tasmota 14.1 with last year's core but don't know how to build with last year's core. Maybe something has changed in core and not tasmota

**Reply by sfromis** [reply]
If you do not need a custom build, you can just follow the version-specific links at the bottom of the regular OTA page. https://ota.tasmota.com/tasmota/release/

**Reply by gg23303013** [reply]
Sorry for my mistake. I checked again. SDK 2.7.6 is not working.
Working: 12.5.0.3 (tasmota-4M), SDK: 2_7_4_9/2.2.2-dev (38a443e)
Not working: 12.5.0.3 (tasmota-4M), SDK: 2_7_7/2.2.2-dev (38a443e)
Can you provide the Tasmota 14.1, SDK: 2_7_4_9/2.2.2-dev (38a443e) bin file?
The new board seems to be picky.

**Reply by Jason2866** [reply]
No, since the Arduino core is the litterally the same. Only never used libraries of the core where removed with changed higher version numbers.
The device is picky necause of bad hardware. Older Tasmota version has less features and "stresses" the device less.

**Reply by gg23303013** [reply]
@Jason2866,I use VSCode/PlatformIO. How can I downgrade the core version from 2.7.7 to 2.7.4? After rebuilding, my custom Tasmota 12.5 no longer works.  The new module with a chip/antenna on board is neater and cheaper and will be popular soon.

**Reply by sfromis** [reply]
Your story is a good reason for that board to not become popular, at it seems to be bad quality.

**Reply by Jason2866** [reply]
Checkout the git Tasmota git commit in branch development from the date you think it works.
Compile as it is. We do **NOT** support any kind of mixing old and new stuff. You walk alone when doing this. Honestly do yourself a favour and do not use this piece of hardware.

**Reply by sfromis** [reply]
If your crappy board seems to work better with the old version, just use an old version of Tasmota having that. While Tasmota certainly has improvements with versions, older versions works fine too. If you want to spend time on taking a deep dive to mix and match for a different build, you're free to do that - on your own.

**Reply by gg23303013** [reply]
https://www.ebay.com/itm/126054221780?chn=ps&norover=1&mkevt=1&mkrid=711-166974-028196-7&mkcid=2&mkscid=101&itemid=126054221780&targetid=2290064846347&device=m&mktype=pla&googleloc=9040368&poi=&campaignid=20884184293&mkgroupid=160215073893&rlsatarget=pla-2290064846347&abcId=9360014&merchantid=527396065&geoid=9040368&gad_source=1&gclid=CjwKCAjwzIK1BhAuEiwAHQmU3qnHRSLUVnz217JapWe5ty8QYQTHpilWxYwwlqh8eus3eLzR7r07ahoC0PYQAvD_BwE
Link NodeMCU without ESP12. The board with unsoldered pins can be replaced with male ,female headers or directly wire. I am new. Can you teach how to build with chosen core version. I use custom USER_BACKLOG. New rebuild, new software , module stop working -> throw away the hardware? @@

**Reply by Jason2866** [reply, marked as answer]
Yes, throw away this piece of defect by design hardware. No teaching in how Tasmota is compiled and all together is working. It is a DIY Open Source project. Documentation is here: https://tasmota.github.io/docs/

**Reply by gg23303013** [reply]
I soldered the ULN2803 directly onto the module.

**Reply by sfromis** [reply]
Your board is not even really cheaper than a NodeMCU with a "real" ESP-12 instead, saw suggestions on the same page. I'd prefer the latter....
